### DISCUSSION TITLE
Is there a way to use Linux home directory or cwd in lredir command?

### DISCUSSION THREAD

**Discussion by ecm-pushbx** [dosemu2/dosemu2#2078] [Q&A]
I'd like to write a batch file that would redirect based on a directory in the current working directory of the user.

Eg if cwd is `/home/user/dir` instead of `lredir i: /home/user/dir/subdir` I want a command that locates the directory based on specifying just "subdir".

**Comment by ecm-pushbx** [comment]
Ah I found I can do `emudrv -w` to get the cwd displayed to DOS. I think FreeCOM allows to capture the stdout into a variable. https://github.com/dosemu2/dosemu2/blob/1ca33160b6955813a4e7b67b52e6393fac0a7065/src/dosext/builtins/lredir.c#L563

**Comment by stsp** [comment]
Use unix -w as for emudrv this is iirc
undocumented.

**Reply by ecm-pushbx** [reply]
Ah, and `unix -w` works even for the 2019 revision 9cf2b7560. Thanks!

**Comment by stsp** [comment]
OMG, my memory serves me badly.
`unix -w` gives you unix cwd.
`emudrv -w` gives you unix path of
the DOS cds.

**Reply by ecm-pushbx** [reply]
Ah okay, well they're both fine for me.

**Reply by stsp** [reply]
Have you tried `~` as a home
dir prefix?

**Reply by ecm-pushbx** [reply]
Doesn't work on older dosemu2 9cf2b7560:

```
Welcome to dosemu2!
    Build 2.0pre8-20191102-1332-g9cf2b7560
C:\>lredir i: ~/test
Error 1 (invalid argument) while redirecting drive I: to ~/TEST
C:\>
```

On 4beffeff4 the same command works.

**Reply by ecm-pushbx** [reply]
Same for a path of just `.` - On 4beffef it works (presumably from Linux cwd), on 9cf2b75 it doesn't work.

**Reply by stsp** [reply]
Yes, many things do not work on
a 4 years old commit.
What's the point?
You wanted to point that dosemu2
improves? :)

**Reply by ecm-pushbx** [reply]
In part that, yes! Another part is I don't want to risk breaking things in new ways which they won't if I don't update the default dosemu2 install.

**Reply by stsp** [reply]
Are you running dosemu2 on a
production server or what?
All breakages are fixed in a days,
as you know from your experience here.
And IMHO this is a good model, unless
someone runs dosemu2 in a production.
Quite surprisingly I know a few ppl do
just that (on their own risk, I never
recommend that), but its not you AFAIK.

**Reply by ecm-pushbx** [reply]
Some of my automatic (up to) daily current release builds require a dosemu for parts of their build process. (EDR-DOS that is soon to be added for all assembling, compiling, and linking. lDebug/RxDOS `mak.sh` needs to run the decompression test to determine the exact needed `exeMinAlloc` value for the compressed executables. WarpLink is also not added yet but will need to run the final linking step (linked by itself) in a DOS.)

So, yes, it sort of is a "production" system.

**Reply by ecm-pushbx** [reply]
I forgot MSDebug which I think to recall also assembles and links all in DOS. (Using the "free but no source" MASM and linker included with the 2018 MS-DOS v2 release.)

**Reply by stsp** [reply]
Is this all the ecosystem around idebug?

**Reply by ecm-pushbx** [reply]
lDebug, with a small L. But no, RxDOS, EDR-DOS, MSDebug, and WarpLink are not directly related to the debugger. (ldosboot and inicomp are directly used to build the bootable debugger and compress the triple-mode executable though.)
